CONTRIBUTING.md

```diff
@@ -116,62 +113,34 @@ Every contract MUST implement their corresponding interface that includes all ex
 
-## Testing
+New features must be merged with associated tests. Bug fixes should have a corresponding test that fails without the bug fix.
 
-### Deployer Template
+### Gas Metering
 
-This repo provides a deployer template library for consistency between scripts and unit tests. For more information on how to use the template, check [here](https://github.com/0xPolygon/deployer-kit#readme).
+The [Forge Gas Snapshot](https://github.com/marktoda/forge-gas-snapshot) library is used to measure the gas cost of individual actions. To ensure that the measured gas is accurate, tests have to be run using the isolate argument to generate the correct snapshot and ensure that CI passes:
 
-## Deployment
+```sh
+$ forge test --isolate
+```
 
-This repo utilizes versioned deployments. Any changes to a contract should update the version of this specific contract. A script is provided that extracts deployment information from the `run-latest.json` file within the `broadcast` directory generated while the forge script runs. From this information a JSON and markdown file is generated containing various information about the deployment itself as well as past deployments.
+When adding new functionality, a new gas snapshot should be added, preferably using `snapLastCall`.
 
-### Deployer Template
+## Deployment
 
-This repo provides a deployer template library for consistency between scripts and unit tests. For more information on how to use the template, check [here](https://github.com/0xPolygon/deployer-kit#readme).
+After deployments are executed a script is provided that extracts deployment information from the `run-latest.json` file within the `broadcast` directory generated while the forge script runs. From this information a JSON and markdown file is generated using the [Forge Chronicles](https://github.com/0xPolygon/forge-chronicles) library containing various information about the deployment itself as well as past deployments.
 
 ### Deployment
 
-This repo set up the following RPCs in the `foundry.toml` file:
-
-- mainnet: Ethereum Mainnet
-- goerli: Ethereum Goerli
-- sepolia: Ethereum Sepolia
-- polygon_pos: Polygon PoS
-- mumbai: Polygon Mumbai
-- polygon_zkevm: Polygon zkEVM
-- polygon_zkevm_testnet: Polygon zkEVM Testnet
-
 To deploy the contracts, provide the `--broadcast` flag to the forge script command. Should the etherscan verification time out, it can be picked up again by replacing the `--broadcast` flag with `--resume`.
 Deploy the contracts to one of the predefined networks by providing the according key with the `--rpc-url` flag. Most of the predefined networks require the `INFURA_KEY` environment variable to be set in the `.env` file.
 Including the `--verify` flag will verify deployed contracts on Etherscan. Define the appropriate environment variable for the Etherscan api key in the `.env` file.
 
 ```shell
 forge script script/Deploy.s.sol --broadcast --rpc-url <rpc_url> --verify
```

Comment:
Would maybe add a section on `requirements for merge`?

ie all contributors need signed commits and 1 approval from a maintainer

Reply:
especially good for external contributors